xarray/core/indexing.py

```diff
@@ -850,6 +866,25 @@ def __setitem__(self, key, value):
                         'into memory explicitly using the .load() '
                         'method or accessing its .values attribute.')
 
+    def rolling_window(self, axis, window):
+        """
+        Make an ndarray with a rolling window of axis-th dimension.
+        The rolling dimension will be placed at the last dimension.
+        """
+        import dask.array as da
+
+        if window < 1:
+            raise ValueError(
+                "`window` must be at least 1. Given : {}".format(window))
+        if window > self.array.shape[axis]:
+            raise ValueError("`window` is too long. Given : {}".format(window))
+
+        axis = nputils._validate_axis(self.array, axis)
+        size = self.array.shape[axis] - window + 1
+        arrays = [self.array[(slice(None), ) * axis + (slice(w, size + w), )]
+                  for w in range(window)]
+        return da.stack(arrays, axis=-1)
```

Comment:
Probably the most efficient way to do this would be to use dask's ghost cell support, but this is fine for now:
http://dask.pydata.org/en/latest/array-ghost.html

You would want to map the new efficient rolling window computation over each block.

Reply:
see here as well: https://github.com/pydata/xarray/blob/f3deb2f2495220af819021b199a5305b0d62ef36/xarray/core/dask_array_ops.py#L11-L25

Reply:
Thank you for the help.
But I don't yet come up with the solution for dask...
(Do you mean that we want to do `as_strided`-like operation for each ghosted-chunk?)

I think this could be in another PR, as it would take for along time for me to think of the correct path.